### PR TITLE
Add support for non-public OIDC clients

### LINKED ISSUE #1261: Add support for non public OIDC client
**Description**

Allow for the usage of non-public OIDC clients.

**Reasons**
Currently, the busola dashboard only supports the usage of public OIDC clients. We however are using a private OIDC client for our authentication.  

I have implemented it in our fork of the dashboard by basically adding `clientSecret` in all places where `clientId` is referenced, and it works perfectly. However, @valentinvieriu mentioned the framework should have support for this already. So maybe my implementation is not really in line with how the luigi framework handles things.  

I am happy to create a PR with my implementation, if it is the correct way of doing it. Here are some snippets:

**Attachments**
Screenshot 2022-04-19 at 12 32 59
```javascript
// core/src/luigi-config/auth/oidc-params.js

const OIDC_PARAM_NAMES = new Map([
  ['--oidc-issuer-url', 'issuerUrl'],
  ['--oidc-client-id', 'clientId'],
  ['--oidc-client-secret', 'clientSecret'],
  ['--oidc-extra-scope', 'scope'],
]);

export function parseOIDCParams({ exec: commandData }) {
  if (!commandData || !commandData.args) throw new Error('No args provided');
  let output = {};

  commandData.args.forEach(arg => {
    const [argKey, argValue] = arg.split('=');
    if (!OIDC_PARAM_NAMES.has(argKey)) return;

    const outputKey = OIDC_PARAM_NAMES.get(argKey);
    if (output[outputKey]) output[outputKey] += ' ' + argValue;
    else output[outputKey] = argValue;
  });

  return output;
}

export function createLoginCommand(oidcConfig) {
  return {
    apiVersion: 'client.authentication.k8s.io/v1beta1',
    command: 'kubectl',
    args: [
      'oidc-login',
      'get-token',
      `--oidc-issuer-url=${oidcConfig.issuerUrl}`,
      `--oidc-client-id=${oidcConfig.clientId}`,
      `--oidc-client-secret=${oidcConfig.clientSecret}`,
      `--oidc-extra-scope=openid ${oidcConfig.scope}`,
      '--grant-type=auto',
    ],
  };
}

```

### PR DESCRIPTION
**Pull request by KaiReichart**
**Description**

This PR adds support for non-public OIDC Clients.  

Changes proposed in this pull request:

- add optional clientSecret everywhere where clientId is mentioned
- add clientSecret to tests
- Updated documentation

**Related issue(s)**
Resolves #1261

### CODE REVIEW

**Comment by KaiReichart** [comment]
I added the PR tag to frontend and backend deployment, not 100% sure if it is correct for the tests, let me know if i need to change it

File: core-ui/src/components/Clusters/views/EditCluster/EditCluster.js [hunk 1]
```diff
@@ -74,9 +74,12 @@ function EditClusterComponent({
     findInitialValue('oidc-issuer-url'),
   );
   const [clientId, setClientId] = useState(findInitialValue('oidc-client-id'));
+  const [clientSecret, setClientSecret] = useState(
+    findInitialValue('oidc-client-id'),
+  );
```
Comment (Lyczeq) [inline core-ui/src/components/Clusters/views/EditCluster/EditCluster.js:79]: I think there should be `oidc-client-secret`.
```suggestion
  const [clientSecret, setClientSecret] = useState(
    findInitialValue('oidc-client-secret'),
  );
```

Comment (KaiReichart) [reply core-ui/src/components/Clusters/views/EditCluster/EditCluster.js:79]: omg yes of course, good catch

**Review by dariadomagala — CHANGES_REQUESTED** [review]
One small suggestion, other that that, it looks good

File: core-ui/src/components/Clusters/components/AuthForm.js [hunk 2]
```diff
@@ -47,6 +47,13 @@ const OIDCform = ({ resource, setResource, ...props }) => {
         input={Inputs.Text}
         aria-label="client-id"
       />
+      <ResourceForm.FormField
+        required
```
[the remark this answers is not in the document]
Comment (KaiReichart) [reply core-ui/src/components/Clusters/components/AuthForm.js]: removed it, thx

**Review by dariadomagala — DISMISSED** [review]

File: docs/features.md [hunk 3]
```diff
@@ -346,15 +346,16 @@ The **match** keys and **messageSrc** must use the format described in the [`jso
   },
   ```
 
-- **SSO_LOGIN** – is used to configure data necessary for the SSO login such as an issuer address, client’s ID, and scopes.
+- **SSO_LOGIN** – is used to configure data necessary for the SSO login such as an issuer address, client’s ID, client’s Secret and scopes. If clientSecret is omitted, a public client is assumed.
```
Comment (grego952) [inline docs/features.md:349]:
```suggestion
- **SSO_LOGIN** – is used to configure data necessary for the SSO login such as an issuer address, client’s ID, client’s Secret and scopes. If the client's Secret is omitted, a public client is used.
```

Comment (Wawrzyn321) [reply docs/features.md:349]: I'd go with `clientSecret`, as it's a name of the variable, not some client's K8s Secret resource.
```suggestion
- **SSO_LOGIN** – is used to configure data necessary for the SSO login such as an issuer address, client’s ID, client’s Secret and scopes. If `clientSecret` is omitted, a public client is used.
```
I've added backticks to emphasize that.

**Review by Wawrzyn321 — APPROVED** [review]

**Review by grego952 — APPROVED** [review]

**Pull request merged**